From: Richard Acayan <mailingradian@gmail.com>
Date: Mon, 5 Dec 2022 18:07:49 -0500
Subject: [PATCH] dt-bindings: soc: qcom: add modemsmem

Some modems require a SMEM client to give them information about the
SoC (version, platform type, and platform subtype). Add schema for the
client.

Signed-off-by: Richard Acayan <mailingradian@gmail.com>
---
 .../bindings/soc/qcom/google,modemsmem.yaml        | 40 ++++++++++++++++++++++
 1 file changed, 40 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/soc/qcom/google,modemsmem.yaml

diff --git a/Documentation/devicetree/bindings/soc/qcom/google,modemsmem.yaml b/Documentation/devicetree/bindings/soc/qcom/google,modemsmem.yaml
new file mode 100644
index 0000000..59fa8a9
--- /dev/null
+++ b/Documentation/devicetree/bindings/soc/qcom/google,modemsmem.yaml
@@ -0,0 +1,40 @@
+# SPDX-License-Identifier: GPL-2.0-only OR BSD-2-Clause
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/soc/qcom/google,modemsmem.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: SoC Information over Shared Memory for Google Pixel Modems
+
+maintainers:
+  - Richard Acayan <mailingradian@gmail.com>
+
+description:
+  The modemsmem client provides information about the SoC to the modem over a
+  vendor-reserved Shared Memory (SMEM) interface.
+
+properties:
+  compatible:
+    const: google,modemsmem
+
+  qcom,smem:
+    $ref: /schemas/types.yaml#/definitions/uint32-array
+    items:
+      minItems: 1
+      maxItems: 1
+    description:
+      Shared memory interface over which the SoC information should be
+      populated.
+
+required:
+  - compatible
+  - qcom,smem
+
+unevaluatedProperties: false
+
+examples:
+  - |
+    modemsmem {
+        compatible = "google,modemsmem";
+        qcom,smem = <134>;
+    };
